src/metric.rs

```diff
@@ -77,7 +79,7 @@ pub trait Metric: Monoid + Eq + Add<usize, Output=Self>
 }
 
 #[derive(Clone, Copy, Debug, PartialOrd, Ord, PartialEq, Eq)]
-pub struct Grapheme(usize);
+pub struct Grapheme(pub usize);
 impl Default for Grapheme {
```

Comment:
What do you think about a `zero` method for these structs instead of just `default`? This is mostly bikeshed but I think it's clearer.